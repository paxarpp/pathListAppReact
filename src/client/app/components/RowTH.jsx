import React, { Component } from 'react';
import PropTypes from 'prop-types';

import fildNamePathList from './fildNamePathList';
import SortLink from './SortLink';

export default class RowTH extends Component {
  handler = e => {
    const { handlerTable } = this.props;
    handlerTable(e.target.dataset.fieldName);
  };
  render() {
    const { reverse, name } = this.props;
    return (
      <tr onClick={this.handler}>
        <th data-field-name={'dateBegin'} style={{ width: '10%' }}>
          {fildNamePathList['dateBegin']}
          <SortLink name={name} nameCell={'dateBegin'} reverse={reverse} />
        </th>
        <th style={{ width: '5%' }}>{fildNamePathList['fuel']}</th>
        <th style={{ width: '5%' }}>{fildNamePathList['constFuelChange']}</th>
        <th data-field-name={'pathBegin'} style={{ width: '10%' }}>
          {fildNamePathList['pathBegin']}
          <SortLink name={name} nameCell={'pathBegin'} reverse={reverse} />
        </th>
        <th data-field-name={'pathEnd'} style={{ width: '10%' }}>
          {fildNamePathList['pathEnd']}
          <SortLink name={name} nameCell={'pathEnd'} reverse={reverse} />
        </th>
        <th data-field-name={'milleage'} style={{ width: '10%' }}>
          {fildNamePathList['milleage']}
          <SortLink name={name} nameCell={'milleage'} reverse={reverse} />
        </th>
        <th data-field-name={'fuelBegin'} style={{ width: '10%' }}>
          {fildNamePathList['fuelBegin']}
          <SortLink name={name} nameCell={'fuelBegin'} reverse={reverse} />
        </th>
        <th data-field-name={'addFuel'} style={{ width: '10%' }}>
          {fildNamePathList['addFuel']}
          <SortLink name={name} nameCell={'addFuel'} reverse={reverse} />
        </th>
        <th data-field-name={'deltaFuel'} style={{ width: '10%' }}>
          {fildNamePathList['deltaFuel']}
          <SortLink name={name} nameCell={'deltaFuel'} reverse={reverse} />
        </th>
        <th data-field-name={'fuelEnd'} style={{ width: '10%' }} colSpan={2}>
          {fildNamePathList['fuelEnd']}
          <SortLink name={name} nameCell={'fuelEnd'} reverse={reverse} />
        </th>
      </tr>
    );
  }
}
RowTH.propTypes = {
  handlerTable: PropTypes.func,
  reverse: PropTypes.bool,
  name: PropTypes.string
};
