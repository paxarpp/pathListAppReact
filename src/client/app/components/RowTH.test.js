import React from 'react';
import renderer from 'react-test-renderer';
import RowTH from './RowTH';

test('RowTH view, reverse false not name', () => {
  const reverse = false;
  const name = '';
  const component = renderer.create(<RowTH handlerTable={() => {}} reverse={reverse} name={name} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('RowTH view, reverse false', () => {
  const reverse = false;
  const name = 'dateBegin';
  const component = renderer.create(<RowTH handlerTable={() => {}} reverse={reverse} name={name} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('RowTH view, reverse true', () => {
  const reverse = false;
  const name = 'dateBegin';
  const component = renderer.create(<RowTH handlerTable={() => {}} reverse={reverse} name={name} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
