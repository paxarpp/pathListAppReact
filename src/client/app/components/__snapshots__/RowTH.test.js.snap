// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RowTH view, reverse false 1`] = `
<tr
  onClick={[Function]}
>
  <th
    data-field-name="dateBegin"
    style={
      Object {
        "width": "10%",
      }
    }
  >
    дата
    <a
      className="sortView"
      onClick={null}
    >
      ▲
    </a>
  </th>
  <th
    style={
      Object {
        "width": "5%",
      }
    }
  >
    тип топлива
  </th>
  <th
    style={
      Object {
        "width": "5%",
      }
    }
  >
    паспортный расход на 100км
  </th>
  <th
    data-field-name="pathBegin"
    style={
      Object {
        "width": "10%",
      }
    }
  >
    пробег в начале
    <div
      className="iconSort"
    >
      <div>
        ▲
      </div>
      <div>
        ▼
      </div>
    </div>
  </th>
  <th
    data-field-name="pathEnd"
    style={
      Object {
        "width": "10%",
      }
    }
  >
    пробег в конце
    <div
      className="iconSort"
    >
      <div>
        ▲
      </div>
      <div>
        ▼
      </div>
    </div>
  </th>
  <th
    data-field-name="milleage"
    style={
      Object {
        "width": "10%",
      }
    }
  >
    пробег
    <div
      className="iconSort"
    >
      <div>
        ▲
      </div>
      <div>
        ▼
      </div>
    </div>
  </th>
  <th
    data-field-name="fuelBegin"
    style={
      Object {
        "width": "10%",
      }
    }
  >
    топливо в начале
    <div
      className="iconSort"
    >
      <div>
        ▲
      </div>
      <div>
        ▼
      </div>
    </div>
  </th>
  <th
    data-field-name="addFuel"
    style={
      Object {
        "width": "10%",
      }
    }
  >
    заправлено
    <div
      className="iconSort"
    >
      <div>
        ▲
      </div>
      <div>
        ▼
      </div>
    </div>
  </th>
  <th
    data-field-name="deltaFuel"
    style={
      Object {
        "width": "10%",
      }
    }
  >
    расход топлива за пробег
    <div
      className="iconSort"
    >
      <div>
        ▲
      </div>
      <div>
        ▼
      </div>
    </div>
  </th>
  <th
    colSpan={2}
    data-field-name="fuelEnd"
    style={
      Object {
        "width": "10%",
      }
    }
  >
    топливо в конце
    <div
      className="iconSort"
    >
      <div>
        ▲
      </div>
      <div>
        ▼
      </div>
    </div>
  </th>
</tr>
`;

exports[`RowTH view, reverse false not name 1`] = `
<tr
  onClick={[Function]}
>
  <th
    data-field-name="dateBegin"
    style={
      Object {
        "width": "10%",
      }
    }
  >
    дата
    <div
      className="iconSort"
    >
      <div>
        ▲
      </div>
      <div>
        ▼
      </div>
    </div>
  </th>
  <th
    style={
      Object {
        "width": "5%",
      }
    }
  >
    тип топлива
  </th>
  <th
    style={
      Object {
        "width": "5%",
      }
    }
  >
    паспортный расход на 100км
  </th>
  <th
    data-field-name="pathBegin"
    style={
      Object {
        "width": "10%",
      }
    }
  >
    пробег в начале
    <div
      className="iconSort"
    >
      <div>
        ▲
      </div>
      <div>
        ▼
      </div>
    </div>
  </th>
  <th
    data-field-name="pathEnd"
    style={
      Object {
        "width": "10%",
      }
    }
  >
    пробег в конце
    <div
      className="iconSort"
    >
      <div>
        ▲
      </div>
      <div>
        ▼
      </div>
    </div>
  </th>
  <th
    data-field-name="milleage"
    style={
      Object {
        "width": "10%",
      }
    }
  >
    пробег
    <div
      className="iconSort"
    >
      <div>
        ▲
      </div>
      <div>
        ▼
      </div>
    </div>
  </th>
  <th
    data-field-name="fuelBegin"
    style={
      Object {
        "width": "10%",
      }
    }
  >
    топливо в начале
    <div
      className="iconSort"
    >
      <div>
        ▲
      </div>
      <div>
        ▼
      </div>
    </div>
  </th>
  <th
    data-field-name="addFuel"
    style={
      Object {
        "width": "10%",
      }
    }
  >
    заправлено
    <div
      className="iconSort"
    >
      <div>
        ▲
      </div>
      <div>
        ▼
      </div>
    </div>
  </th>
  <th
    data-field-name="deltaFuel"
    style={
      Object {
        "width": "10%",
      }
    }
  >
    расход топлива за пробег
    <div
      className="iconSort"
    >
      <div>
        ▲
      </div>
      <div>
        ▼
      </div>
    </div>
  </th>
  <th
    colSpan={2}
    data-field-name="fuelEnd"
    style={
      Object {
        "width": "10%",
      }
    }
  >
    топливо в конце
    <div
      className="iconSort"
    >
      <div>
        ▲
      </div>
      <div>
        ▼
      </div>
    </div>
  </th>
</tr>
`;

exports[`RowTH view, reverse true 1`] = `
<tr
  onClick={[Function]}
>
  <th
    data-field-name="dateBegin"
    style={
      Object {
        "width": "10%",
      }
    }
  >
    дата
    <a
      className="sortView"
      onClick={null}
    >
      ▲
    </a>
  </th>
  <th
    style={
      Object {
        "width": "5%",
      }
    }
  >
    тип топлива
  </th>
  <th
    style={
      Object {
        "width": "5%",
      }
    }
  >
    паспортный расход на 100км
  </th>
  <th
    data-field-name="pathBegin"
    style={
      Object {
        "width": "10%",
      }
    }
  >
    пробег в начале
    <div
      className="iconSort"
    >
      <div>
        ▲
      </div>
      <div>
        ▼
      </div>
    </div>
  </th>
  <th
    data-field-name="pathEnd"
    style={
      Object {
        "width": "10%",
      }
    }
  >
    пробег в конце
    <div
      className="iconSort"
    >
      <div>
        ▲
      </div>
      <div>
        ▼
      </div>
    </div>
  </th>
  <th
    data-field-name="milleage"
    style={
      Object {
        "width": "10%",
      }
    }
  >
    пробег
    <div
      className="iconSort"
    >
      <div>
        ▲
      </div>
      <div>
        ▼
      </div>
    </div>
  </th>
  <th
    data-field-name="fuelBegin"
    style={
      Object {
        "width": "10%",
      }
    }
  >
    топливо в начале
    <div
      className="iconSort"
    >
      <div>
        ▲
      </div>
      <div>
        ▼
      </div>
    </div>
  </th>
  <th
    data-field-name="addFuel"
    style={
      Object {
        "width": "10%",
      }
    }
  >
    заправлено
    <div
      className="iconSort"
    >
      <div>
        ▲
      </div>
      <div>
        ▼
      </div>
    </div>
  </th>
  <th
    data-field-name="deltaFuel"
    style={
      Object {
        "width": "10%",
      }
    }
  >
    расход топлива за пробег
    <div
      className="iconSort"
    >
      <div>
        ▲
      </div>
      <div>
        ▼
      </div>
    </div>
  </th>
  <th
    colSpan={2}
    data-field-name="fuelEnd"
    style={
      Object {
        "width": "10%",
      }
    }
  >
    топливо в конце
    <div
      className="iconSort"
    >
      <div>
        ▲
      </div>
      <div>
        ▼
      </div>
    </div>
  </th>
</tr>
`;
